docs/devguide/contributing.rst

```diff
@@ -35,3 +35,86 @@ negative votes, the team will grant commit privileges.
+will run automatically, test results can be found in the merge status field of
+each pull request page.
+
+.. _Hound: https://houndci.com/
+.. _Travis CI: https://travis-ci.org/
+.. _Jenkins: https://jenkins.io/
+.. _jshint: http://jshint.com/
+
+**Hound**
+
+Hound works with `jshint`_ and comments on style violations in pull requests. 
+Configuration files ``.hound.yml`` and ``.jshintrc`` have been created in each 
+repository, so before creating a pull request, you can check code style locally with 
+jshint to find out style violations beforehand.
+
+**Travis CI**
+
+Travis CI runs the unit tests, and then does some potentially ancillary actions.
+The build specifics are detailed in the ``.travis.yml`` file within each repository.
+For finding out basic errors before creating a pull request, you can run unit test
+locally using ``npm test`` within each repository.
+
+**Jenkins**
+
+Jenkins uses the Github Pull Request Builder plugin to monitor all pull requests
+to perform quality gate tests prior to merge. The gates include running all the unit
+tests, running all dependent project unit tests with the code proposed from the pull 
+request, running an integration "smoke test" to verify basic end to end functionality
+and commenting on the details of test case failure. Jenkins is also able to takes 
+instructions from pull request comments to deal with complex test scenarios. 
+Instructions can be written in pull request description or comments.
+
+The following table show all the Jenkins Instructions and usage:
+
+.. list-table::
+    :widths: 30 50 100 
+    :header-rows: 1
+
+    * - Instruction
+      - Description
+      - Detailed Usage
+    * - Jenkins: test this please
+      - Trigger one Jenkins test on current pull request.
+      - This command will trigger one Jenkins test no matter whether the pull request has been
+        tested or not, the new test result will cover the previous one. 
+    * - Jenkins: ignore
+      - Avoid running any test in the next Jenkins test.
+      - This command can be used in "work in progress" pull request or one of 
+        "interdependent pull requests" to avoid a Jenkins test that is destined to fail.
+        The commit status of this pull request will be set to "pending" temporally.
+        Jenkins periodically check the repository status so to guarantee to avoid the 
+        unexpected test, it's strongly advised to write this command in pull request description.
+        Otherwise if write in comments, an unexpected test may be triggered in the time slot
+        between pull request creation and writing up comments. 
+
+        Remove this instruction and then "test this please" can recovery Jenkins test on this pull request 
+        and the commit status "pending" will be replaced by new result. If one pull request is depended on 
+        by another, the result of the combination test will replace "pending" too.
+        
+    * - Jenkins: depends on pr1_url, pr2_url ...
```

Comment:
Recommends to add few real life example on how should pr1_url1 or pr2_url2 be writen.